### ISSUE TITLE
What does `demo_postprocess()` do?

### ISSUE THREAD

**Issue by tensorturtle** [Megvii-BaseDetection/YOLOX#690]
I need help understanding what [utils/demo_utils.py line 99](https://github.com/Megvii-BaseDetection/YOLOX/blob/main/yolox/utils/demo_utils.py) does.

Why is this postprocessing step required?

How does it transform the output?

If I know that my image size and P6 will be same forever, would it be possible to pre-calculate some of the expressions in this function and therefore make it faster?

**Comment by Joker316701882** [comment]
@tensorturtle The regression output from the model is the regression offset, we need to decode it into the box.

"If I know that my image size and P6 will be same forever, would it be possible to pre-calculate some of the expressions in this function and therefore make it faster?"

Knowing not using P6 will not increase the speed.

**Comment by tensorturtle** [comment]
@Joker316701882 Thank you for your kind explanation.
I'm still trying to understand what 'decoding the regression offset' means...
Is there any way you could explain it in simpler terms?
